### PR TITLE
Added ride passenger fee price calculation

### PR DESCRIPTION
**Pull request by erezdrutin**
image
image

### CODE REVIEW

**Comment by erezdrutin** [comment]
(we need to change the image lmfao)

File: apps/BE/src/controllers/trip/types.ts [hunk 1]
```diff
@@ -1,27 +1,36 @@
 export interface Address {
-    location_id?: string;
-    lat: number;
-    lon: number;
+  location_id?: string;
+  lat: number;
+  lon: number;
```
Comment (Zoeshwa) [inline apps/BE/src/controllers/trip/types.ts:4]: Don't you need a name too?

File: apps/BE/src/controllers/trip/types.ts [hunk 2]
```diff
@@ -1,27 +1,36 @@
...
-            activities: Array<{
-                address: Address;
-            }>;
-        }>;
-    };
+  solution: {
+    routes: Array<{
+      activities: Array<{
+        address: Address;
+      }>;
```
Comment (Zoeshwa) [inline apps/BE/src/controllers/trip/types.ts:24]: Use [] instead

File: apps/ui/src/App.tsx [hunk 3]
```diff
@@ -41,7 +41,7 @@ const App: React.FC = () => {
                         element={<CommunitiesFeed />}
                       />
                       <Route path="/home" element={<HomePage />} />
-                      <Route path="/map/rides/:rideId" element={<MapRouting />} />
+                      <Route path="/rides/:rideId" element={<MapRouting />} />
```
Comment (Zoeshwa) [inline apps/ui/src/App.tsx:44]: Please make sure it is supported in both versions (mobile & desktop)

File: apps/ui/src/Pages/MapRouting/MapRoutingMachine.tsx [hunk 4]
```diff
@@ -1,26 +1,39 @@
...
+import { ButtonGroup, Button, Tooltip } from '@mui/material';
+import { PriceChange as PaymentIcon } from '@mui/icons-material';
+import { RideFeeDialog, RideFeeDialogHandle } from './RideFeeDialog';
 
 function MapRouting() {
     const { rideId } = useParams();
-    const { data: locations, isLoading, error } = useGetTrip(Number(rideId));
+    const { data, isLoading, error } = useGetTrip(Number(rideId));
+    const dialogRef = useRef<RideFeeDialogHandle>(null);
+    const pricePerPassenger = 25;  // Define this dynamically based on your application needs
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/MapRouting/MapRoutingMachine.tsx:16]: Save it as a const in the consts file

File: apps/ui/src/Pages/MapRouting/Routing.tsx [hunk 5]
```diff
@@ -5,46 +5,51 @@ import "leaflet-routing-machine";
...
+  if (waypoints[index].type === "start") {
+    return "Starting point";
+  } else if (waypoints[index].type === "end") {
+    return "Drive destination";
+  } else {
+    return `${waypoints[index].userName}'s ${waypoints[index].type}`;
   }
+}
+
+export default function Routing({ waypoints, dialogRef, passengersCount }: RoutingProps) {
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/MapRouting/Routing.tsx:49]: const Routing = () => {}

File: apps/ui/src/Pages/MapRouting/Routing.tsx [hunk 6]
```diff
@@ -68,6 +73,23 @@ export default function Routing({ waypoints }: RoutingProps) {
...
       .addTo(map);
+    
+      // Calculate the fare based on the distance, using the formula presented in https://www.omnicalculator.com/everyday-life/carpooling
+      routingControl.on('routesfound', (e) => {      
+        const routes = e.routes;
+        const summary = routes[0].summary;
+
+        const distanceInKM = summary.totalDistance / 1000;
+        const fuelConsumption = distanceInKM / 6.5; // 6-7 liters per 100km  https://www.carsguide.com.au/car-advice/what-is-average-fuel-consumption-88469#:~:text=However%2C%20as%20a%20rule%20of,100km%20in%20the%20real%20world.
+        const fuelCost = fuelConsumption * 8; // average 8 shekels per liter
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/MapRouting/Routing.tsx:84]: make 8 & 6.5 consts

File: apps/ui/src/Pages/MapRouting/Routing.tsx [hunk 7]
```diff
@@ -68,6 +73,23 @@ export default function Routing({ waypoints }: RoutingProps) {
...
+
+        const rideFee = fuelCost / passengersCount;
+      
+        // Update the fare in the dialogRef current object if it exists
+        if (dialogRef.current) {
+          dialogRef.current.updateFare(Math.ceil(rideFee)); // You'll need to add this method to your DialogHandle interface
```
Comment (Zoeshwa) [inline apps/ui/src/Pages/MapRouting/Routing.tsx:90]: did you added this to the dialog?

**Review by Zoeshwa — COMMENTED** [review]
Cool, I left some comments :)

**Comment by erezdrutin** [comment]
@Kovler10 both builds (BE/FE) work locally, can you please check?

**Pull request merged**